src/climsight/climsight_engine.py

```diff
@@ -370,6 +381,62 @@ def llm_request(content_message, input_params, config, api_key, stream_handler):
     #yield f"Generating..."
     logging.info(f"Generating...")    
 
+    ## === RAG integration === ##
+    # langchain embeddig model
+    langchain_ef = OpenAIEmbeddings(openai_api_key=api_key, model=EMBEDDING_MODEL) # max_retries, request_timeout, retry_min_seconds
+    # load & retrieve database 
+    db = Chroma(persist_directory=CHROMA_PATH, embedding_function=langchain_ef, collection_name="ipcc-collection")
+    print(f"There are {db._collection.count()} documents in the collection.")
```

Comment:
move CHROMA_PATH from global variables to config

Reply:
Move the RAG integration to rag.py as a function.
Do not call this function if RAG has not been initialized. 
Pass api_key, config, and input_params to this function.
Avoid loading the database multiple times; save the database in memory for reuse after the first call (preferably load the database during the initialization phase).

Reply:
Done

Reply:
Done